packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -707,7 +716,7 @@ export default class InternalModel {
   // TODO Igor consider getting rid of initial state
   getManyArray(key, isAsync = false) {
     let relationshipMeta = this.store._relationshipMetaFor(this.modelName, null, key);
-    let jsonApi = this._recordData.getHasMany(key);
+    let jsonApi = (this._recordData as DefaultRecordData).getHasMany(key);
```

Comment:
Same question here and for all similar casts below: how do we know this cast is safe?

Reply:
see above